src/infrasys/system.py

```diff
@@ -193,8 +192,9 @@ def to_json(self, filename: Path | str, overwrite=False, indent=None, data=None)
         backup(self._con, time_series_dir / self.DB_FILENAME)
         self._time_series_mgr.serialize(system_data["time_series"], time_series_dir)
 
-        with open(filename, "w", encoding="utf-8") as f_out:
-            json.dump(data, f_out, indent=indent)
+        data_dump = orjson.dumps(data)
+        with open(filename, "wb") as f_out:
+            f_out.write(data_dump)
```

Comment:
Is there not a `orjson.dump(data, f_out)`? Then you wouldn't have to instantiate the full string in memory.